### PR TITLE
Standardize deploy on the same step system as everything else

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Standardize deploy on the same step system as update code/cloud debug
- Add image based deploy test

## Screenshots
Screen Shot 2020-12-03 at 1 10 19 PM
Screen Shot 2020-12-03 at 1 09 35 PM

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kiiadi** [comment]
how does this work when I want to confirm before i deploy?

**Comment by hunterwerlla** [comment]
> how does this work when I want to confirm before i deploy?

See screenshot 2

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt [hunk 1]
```diff
@@ -41,6 +47,11 @@ class SamDeployTest {
         .region(Region.US_WEST_2)
         .build()
 
+    private val ecrClient = EcrClient.builder()
+        .httpClient(ApacheHttpClient.builder().build())
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployTest.kt:51]: I don't think we do, the existing one above did this and it was a copy paste, removed from both

**Comment by kiiadi** [comment]
> See screenshot 2

we didn't actually allow them to view the change-set previously right? they had to do that themselves in the console right?

as a future PR - I think we should enhance it, run a 'describe-changeset' and actually show a diff in a table of what's going to happen.

**Review by kiiadi — APPROVED** [review]

**Comment by abrooksv** [comment]
Thank you for doing this, happy to see the Step system put to more use and standardizing our code paths.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt [hunk 2]
```diff
@@ -110,46 +118,75 @@ class DeployServerlessApplicationAction : AnAction(
...
-        if (!deployDialog.isOK) return
+        workflow.onSuccess = {
+            runBlocking {
+                val changeSetArn = it.getRequiredAttribute(DeployLambda.CHANGE_SET_ARN)
 
-        val cfnClient = project.awsClient<CloudFormationClient>()
+                if (!stackDialog.autoExecute) {
+                    val response = withContext(edtContext) { UploadFunctionContinueDialog(project, changeSetArn).showAndGet() }
+                    if (!response) {
+                        // TODO telemetry
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt:144]: Did we never have telemetry here?

**Review by abrooksv — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt:144]: Will clarify the message and fix telemetry here

**Pull request merged**